### PR TITLE
zprobe: add configuration via M670

### PR DESCRIPTION
**Pull request by sorki**
Also adds support for storing probe configuration
to config-override via M500

;Probe feedrates slow/fast/return (mm/sec):
M670 S0.50 K2.00 R2.50
;Probe max_z (mm):
M670 Z200.00
;Probe height (mm):
M670 H2.00

Signed-off-by: Richard Marko <rmarko@fedoraproject.org>

Depends on #704.

### CODE REVIEW

**Comment by sorki** [comment]
removed mark_as_taken stuff

File: src/modules/tools/zprobe/ZProbe.cpp [hunk 1]
```diff
@@ -287,17 +287,35 @@ void ZProbe::on_gcode_received(void *argument)
...
+        switch (gcode->m) {
+            case 119:
+                c = this->pin.get();
+                gcode->stream->printf(" Probe: %d", c);
+                gcode->add_nl = true;
+                break;
+
+            case 670:
+                if (gcode->has_letter('S')) this->slow_feedrate = gcode->get_value('S');
+                if (gcode->has_letter('F')) this->fast_feedrate = gcode->get_value('F');
```
Comment (wolfmanjm) [inline src/modules/tools/zprobe/ZProbe.cpp]: cannot use F here as it is globally reserved to mean G0/G1 feedrate which can affect other things.

File: src/modules/tools/zprobe/ZProbe.cpp [hunk 2]
```diff
@@ -287,17 +287,35 @@ void ZProbe::on_gcode_received(void *argument)
...
+                if (gcode->has_letter('H')) this->probe_height = gcode->get_value('H');
+                break;
+
+            case 500: // save settings
+            case 503: // print settings
+                gcode->stream->printf(";Probe feedrates slow/fast/return (mm/sec):\nM670 S%1.2f F%1.2f R%1.2f\n",
+                    this->slow_feedrate, this->fast_feedrate, this->return_feedrate);
+                gcode->stream->printf(";Probe max_z (mm):\nM670 Z%1.2f\n", this->max_z);
+                gcode->stream->printf(";Probe height (mm):\nM670 H%1.2f\n", this->probe_height);
+
```
Comment (wolfmanjm) [inline src/modules/tools/zprobe/ZProbe.cpp]: add comment here to indicate fall through is intended.

File: src/modules/tools/zprobe/ZProbe.cpp (hunk 1 above)
Comment (sorki) [inline src/modules/tools/zprobe/ZProbe.cpp]: Your suggestion? I can't come up with better fitting letter :)

**Comment by sorki** [comment]
Now using T instead of F, added note.
"; Probe feedrates Slow/fasT/Return (mm/sec):\nM670 S%1.2f T%1.2f R%1.2f\n",

Tested with todays edge, ZGrid works too.

**Comment by wolfmanjm** [comment]
T is also a reserved word.. it is tool change it is not a valid parameter. How about K or Q

**Comment by sorki** [comment]
K sounds good!

**Comment by wolfmanjm** [comment]
Is there a specific reason for splitting the M670 into three lines rather than one? 
I'd prefer to keep it to one line if possible unless there is an overriding reason otherwise.

**Comment by wolfmanjm** [comment]
Never mind I'll change that myself unless you have a valid objection.

**Pull request merged**